conversant-htb.js

```diff
@@ -72,11 +72,124 @@ function ConversantHtb(configs) {
      * @private {object}
      */
     var __bidTransformers;
+    
+    var w = Browser.topWindow;
+    var n = w.navigator;
 
     /* =====================================
      * Functions
      * ---------------------------------- */
 
+    /* Conversant
+     * ---------------------------------- */
+
+    /**
+     * Return the indicator value for do-not-track
+     */
+    
+    function __getDNT() { 
+    	return (n.doNotTrack === '1' || w.doNotTrack === '1' || n.msDoNotTrack === '1' || n.doNotTrack === 'yes') ? 1 : 0;
+    }
+
+    /**
+     * Return the device object for the bid request
+     */
+    
+    function __getDevice() {
+    	return {
+    		h: Browser.getScreenHeight(),
+    		w: Browser.getScreenWidth(),
+    		dnt : __getDNT(),
+    		language: Browser.getLanguage(),
+    		make: n.vendor ? n.vendor : '',
+    		ua: Browser.getUserAgent()
+    	};
+    }
+    
+    /**
+     * Return site object for the bid request
+     */
+    
+    function __getSite() {
+    	return {
+    		id: configs.siteId,
+    		mobile: w.document.querySelector('meta[name="viewport"][content*="width=device-width"]') !== null ? 1 : 0,
+    		page: Browser.getPageUrl()
+    	};
+    }
+    
+    /**
+     * Return an array of impressions for the bid request
+     */
+    
+    function __getImps(returnParcels) {
+    	var conversantImps = [];
+    	var secure = (Browser.getProtocol().search(/^https:/i) >= 0) ? 1 : 0;
+    	var counter = 0;
+
+    	// Each parcel is a unique combination of a htSlot and xSlot.
+    	// Since Conversant bid requests do not require unique placement ids, 
+    	// we create an id based on xSlotName and stash it as cnvrId in the parcel
+    	// so the bid responses and the parcels can be matched up.
+    	// It is always possible to use the requestId in the parcels
+    	// as an alternative approach.
+    	
+    	for (var i = 0; i < returnParcels.length; ++i) {
+    		var parcel = returnParcels[i];
+    		var xSlot = parcel.xSlotRef;
+    		var imp = {};
+    		var banner = {};
+    		var id = parcel.xSlotName + '_' +  counter++;
+    		
+    		parcel.cnvrId = id;
```

Comment:
Hi, can you please switch to the requestId approach? The parcels are used across the whole request response cycle and modification to the parcel object can cause  unwanted collisions.

Reply:
Certainly.  Just want to double check, are requestIds unique?